### ISSUE TITLE
Simulated data

### ISSUE THREAD

**Issue by qtrandev** [adam-s/trolley-tracker#3]
Feature Request:

We often would like to test our maps with simulated movement data from vehicles. Would it be possible to have another endpoint to get this simulated data?

It would probably be the same as the current GET endpoint:
http://trolley-tracker.herokuapp.com/api/simtrolleys?limit=1

You can either use historical data from the last x positions in the database, or record one of the Fort Lauderdale trolleys going in a loop and replaying this data.

**Comment by qtrandev** [comment]
For example, you can record one of the routes here for about 1.5 to 3 hours during the day and replaying that data: http://suntrolley.herokuapp.com/api

**Comment by adam-s** [comment]
We can make http://staging-trolley-tracker.herokuapp.com which gets pushed the code to test in the real world before pushing to production at http://trolley-tracer.herokuapp.com.

**Comment by qtrandev** [comment]
I want the simulated location data. Not really concerned with staging or production code, but you can do it however you like. To me, just another endpoint like /simtrolleys is fine. No need for multiple Heroku apps. Also, don't you only get 5 free apps on Heroku? Just for this, you might be losing 3 of those 5 free apps.

**Comment by adam-s** [comment]
Verified Heroku account get 100 free instances, but they have to sleep for at least 6 hours in a 24 hour period. Also, they shut down after 30 minutes of no use. It takes them a minute or two to wind up again. I'll try with a staging server first and if there is reason to add another end point we can do that instead.

**Comment by qtrandev** [comment]
Try to keep it to 1 app to make it easy for any beginner to try to deploy this app. Also, it would be cheaper if I had to pay to deploy this so it can receive all my GPS tracker data. I might be able to afford 1 app, but it hurts to pay more.

**Comment by qtrandev** [comment]
Basically I can just pay for one dyno to keep this instance running 24 hours a day and all new features go into this app to run on one dyno.

**Comment by adam-s** [comment]
That's fine. So, from this issue I need to add a testing endpoint.

**Comment by qtrandev** [comment]
Yes, everything the same as http://trolley-tracker.herokuapp.com/api/trolleys, but just with historical or saved data that changes just like how buses publish data once in a while.

**Comment by adam-s** [comment]
Thinking out loud... [Use endpoint versioning to manage different endpoint. ](http://stackoverflow.com/questions/26040329/how-do-you-handle-api-version-in-a-node-express-app). It is probably the most elegant solution.

**Comment by adam-s** [comment]
This is frustrating me a bit. I'm just going to quickly add another route and database collection (table) for the test data. Otherwise, there are two good ways to accomplish this requirement. One is create a class to extend because the production code shouldn't have any reference to test code. The best way of course is to have a staging server but this isn't going to happen for reasons discussed above. Sure, it's fine for now, but if the code and system become any more complex than it is now, I highly recommend finding a more sustainable pattern to accomplish this. I'm going to quickly go ahead and add the extra routes, however, we are sacrificing time later for time now. If someone does invest money, quickly setup both a staging server and a mirror production server.

**Comment by qtrandev** [comment]
Low priority item. We don't have to do this. Focus on letting me have a server I can test with.

**Comment by adam-s** [comment]
Try using the server now. Let me know if you have any problems.... we can knock this out very quick.

**Comment by adam-s** [comment]
The production server can be tested and when you are comfortable just use the secret delete token to clear the database to use with real data.

**Comment by qtrandev** [comment]
Some data is flowing up to the server now. More in an hour, but after that will be sporadic.

**Comment by adam-s** [comment]
Cool... let me know how it's working and if any problems come up I'll fix them as fast as I can.
